css/amp.css

```diff
@@ -153,6 +153,14 @@ html.i-amphtml-ios-embed.i-amphtml-ios-overscroll > #i-amphtml-wrapper {
   position: relative;
 }
 
+/**
+ * Children of responsive elements should not be able to influence the height
+ * which they do when they are statically positioned.
+ */
+.i-amphtml-layout-responsive > :not(i-amphtml-sizer) {
```

Comment:
I managed to fix #9397 by using:

```css
[layout]:not(.i-amphtml-element) {
    position: relative;
    overflow: hidden;
    color: transparent !important;
}
[layout]:not(.i-amphtml-element) > * {
    display: none;
}
```

This essentially targets "not built" elements _before_ they're stubbed (once they're stubbed, they're handled by `.i-amphtml-notbuilt`). And it's not affected by SSR!

So we really need:

```diff
- .i-amphtml-notbuilt {
+.i-amphtml-notbuilt, [layout]:not(.i-amphtml-element) {
  position: relative;
  overflow: hidden !important;
  color: transparent !important;
}

- .i-amphtml-notbuilt:not(.i-amphtml-layout-container) > * {
+ .i-amphtml-notbuilt:not(.i-amphtml-layout-container) > *,
+ [layout]:not(.i-amphtml-element) > * {
  display: none;
}